liquidctl/driver/nzxt_smart_device.py

```diff
@@ -182,66 +276,244 @@ def get_status(self, **kwargs):
+                    if ((msg[rpm_offset] != 0x0) and (msg[rpm_offset + 1] != 0x0)):
+                        status.append(('Fan {} speed'.format(i + 1), msg[rpm_offset + 1] << 8 | msg[rpm_offset], 'rpm'))
+                        status.append(('Fan {} duty'.format(i + 1), msg[duty_offset + i], '%'))
+                    rpm_offset += 2
+                status.append(('Noise level', msg[noise_offset], 'dB'))
+                num_valid_replies_recvd += 1
+                msg_6702_reply = True
+                continue
+            if not msg_6704_reply and msg[0] == 0x67 and msg[1] == 0x04:
+            # PLACE HOLDER FOR FUTURE WORK
+            # Interpretation of this reply is pending
+                #status.append((' Unknown:', 'status', 'pending'))
+                num_valid_replies_recvd += 1
+                msg_6704_reply = True
+                continue
+        self.device.release()
+        return sorted(status)
+
+    def _write_colors(self, cid, mode, colors, sval):
+        mval, mod3, mod4, mincolors, maxcolors = self._COLOR_MODES[mode]
+        color_count = len(colors)
+        if maxcolors == 40:
+            channel_mod = [0x1E, 0x20][cid] # the purpose of this is unknown, but is based on cmd issued by CAM software
+            led_padding = [0xFF, 0xFF, 0xFF]*(maxcolors - color_count)  # set all remaining LEDs to White
+            leds = list(itertools.chain(*colors)) + led_padding
+            self._write([0x22, 0x10, cid+1, 0x00] + leds[0:60]) # send first 20 colors to device (3 bytes per color)
+            self._write([0x22, 0x11, cid+1, 0x00] + leds[60:])  # send remaining colors to device
+            msg = self.device.read(self._READ_LENGTH) # wait for one reply before issuing command to specify color mode
+            LOGGER.debug('received %s', ' '.join(format(i, '02x') for i in msg))
+            self._write([0x22, 0xA0, cid+1, 0x00, mval, mod3, 0x00, channel_mod, 0x00,
+                         0x00, 0x64, 0x00, 0x32, 0x00, 0x00, 0x01])
+        elif mode == 'wings':  # wings requires special handling
+            for [g, r, b] in colors:
+                self._write([0x22, 0x10, cid+1])  # clear out all independent LEDs
+                self._write([0x22, 0x11, cid+1])  # clear out all independent LEDs
+                color_list = [g, r, b] * 8
+                r2 = (int)(r // 2.5)
+                g2 = (int)(g // 2.5)
+                b2 = (int)(b // 2.5)
+                color_list2 = [g2, r2, b2] * 8
+                r3 = (int)(r2 // 4)
+                g3 = (int)(g2 // 4)
+                b3 = (int)(b2 // 4)
+                color_list3 = [g3, r3, b3] * 8
+                for i in range(8):   #  send color scheme first, before enabling wings mode
+                    if i == 3 or i == 7:
+                        mod = 0x05
+                    else:
+                        mod = 0x01
+                    msg = ([0x22, 0x20, cid+1, i, 0x04, 0x39, 0x00, mod,
```

Comment:
So if there's any sanity whatsoever left in the firmware developers, this would be equivalent to `mval = 0x04`.  Which would suggest some relation with covering marquees...

Reply:
But the command sequence begins with `\0x22\0x20` instead of `\0x28\0x03`.

Reply:
That's true.